src/domains/ecommerce-common/03_storage.tf

```diff
@@ -391,17 +384,100 @@ resource "azurerm_monitor_scheduled_query_rules_alert" "ecommerce_transient_enqu
   enabled        = true
   query = format(<<-QUERY
     let OpCountForQueue = (operation: string, queueKey: string) {
-      StorageQueueLogs
-      | where OperationName == operation and ObjectKey startswith queueKey
-      | summarize count()
+        StorageQueueLogs
+        | where OperationName == operation and ObjectKey startswith queueKey
+        | summarize count() 
+        | project count_ 
+        | extend dummy=1
+    };
+    let PutMessages = (queueName: string) {
+      OpCountForQueue("PutMessage", queueName)
+        | project PutCount = count_
+        | extend dummy = 1
+    };
+    let DeletedMessages =  (queueName: string) {
+      OpCountForQueue("DeleteMessage", queueName)
+        | project DeleteCount = count_
+        | extend dummy = 1
     };
     let MessageRateForQueue = (queueKey: string) {
-      OpCountForQueue("PutMessage", queueKey)
-      | join kind=fullouter OpCountForQueue("DeleteMessage", queueKey) on count_
-      | project name = queueKey, Count = count_ - count_1
+        PutMessages(queueKey)
+        | join kind=inner (DeletedMessages(queueKey)) on dummy
+        | extend Diff = PutCount - DeleteCount
+        | project PutCount, DeleteCount, Diff
     };
     MessageRateForQueue("%s")
-    | where Count > ${each.value.threshold}
+    | where Diff > ${each.value.threshold}
+    QUERY
+    , "/${module.ecommerce_storage_transient.name}/${local.project}-${each.value.queue_key}"
+  )
+  severity    = each.value.severity
+  frequency   = each.value.frequency
+  time_window = each.value.time_window
+  trigger {
+    operator  = "GreaterThan"
+    threshold = 0
+  }
+}
+
+locals {
+  queue_expiration_alert_props = var.env_short == "p" ? [
+    {
+      "queue_key"   = "transactions-expiration-queue"
+      "severity"    = 1
+      "time_window" = 15
+      "frequency"   = 15
+      "threshold"   = 40
+    },
+  ] : []
+}
+
+# Queue size: Ecommerce - ecommerce queues enqueues rate alert
+resource "azurerm_monitor_scheduled_query_rules_alert" "ecommerce_expiration_enqueue_rate_alert" {
+  for_each            = { for q in local.queue_expiration_alert_props : q.queue_key => q }
+  name                = "${local.project}-${each.value.queue_key}-rate-alert"
+  resource_group_name = azurerm_resource_group.storage_ecommerce_rg.name
+  location            = var.location
+
+  action {
+    action_group           = [data.azurerm_monitor_action_group.email.id, data.azurerm_monitor_action_group.slack.id, azurerm_monitor_action_group.ecommerce_opsgenie[0].id]
+    email_subject          = "Email Header"
+    custom_webhook_payload = "{}"
+  }
+  data_source_id = module.ecommerce_storage_transient.id
+  description    = format("Enqueuing rate for queue %s > ${each.value.threshold} during last ${each.value.time_window} minutes", replace("${each.value.queue_key}", "-", " "))
```

Comment:
Since in the description field we are already using format, maybe we can avoid interpolation. Just an idea:

```description = format("Enqueuing rate for queue %s > %d during last %d minutes", replace(each.value.queue_key, "-", " "), each.value.threshold, each.value.time_window)
`

Reply:
Yes, it's a good idea, but do you mind if we face this point in a dedicated task? If I do it here, I should do it for all alert, and it would be out of the scope. What do you think?

Reply:
yes of course, mine was just an idea, go ahead